### PR TITLE
BACK-822 - Bitcoin explorer service

### PR DESCRIPTION
**Pull request by nukky94**
## What is this about?
- Implement the service to call the bitcoin explorer through http calls

## Cute picture of lama
baby-lama-1-week-old

### CODE REVIEW

File: coins/bitcoin/worker/src/main/resources/application.conf [hunk 1]
```diff
@@ -32,3 +32,9 @@ rabbit {
 
   automatic-recovery = false
 }
+
+explorer {
```
Comment (RomanWlm) [inline coins/bitcoin/worker/src/main/resources/application.conf:36]: Make them ENV configurable :)

**Review by RomanWlm — APPROVED** [review]

File: coins/bitcoin/worker/src/main/scala/co/ledger/lama/bitcoin/worker/App.scala [hunk 2]
```diff
@@ -1,28 +1,58 @@
...
+      rabbitClient <-
+        Resource
+          .make(IO(Executors.newCachedThreadPool()))(es => IO(es.shutdown()))
+          .map(Blocker.liftExecutorService)
+          .evalMap(RabbitClient[IO](conf.rabbit, _))
+
+      httpClient <- BlazeClientBuilder[IO](
+        ExecutionContext.fromExecutorService(Executors.newCachedThreadPool)
+      ).resource
+
```
Comment (vbergeron-ledger) [inline coins/bitcoin/worker/src/main/scala/co/ledger/lama/bitcoin/worker/App.scala:29]: Also this.
it is also better to name those pools to have better error logs.
```suggestion
      httpClient <- Clients.http4s
```

File: coins/bitcoin/worker/src/main/scala/co/ledger/lama/bitcoin/worker/App.scala [hunk 3]
```diff
@@ -1,28 +1,58 @@
...
-      conf.queueName(conf.workerExchangeName),
-      conf.lamaExchangeName,
-      conf.routingKey
-    )
+    val resources = for {
+      rabbitClient <-
+        Resource
+          .make(IO(Executors.newCachedThreadPool()))(es => IO(es.shutdown()))
+          .map(Blocker.liftExecutorService)
+          .evalMap(RabbitClient[IO](conf.rabbit, _))
```
Comment (vbergeron-ledger) [inline coins/bitcoin/worker/src/main/scala/co/ledger/lama/bitcoin/worker/App.scala:24]:
```suggestion
      rabbitClient <- Clients.rabbit(conf.rabbit)
```

Comment (vbergeron-ledger) [reply coins/bitcoin/worker/src/main/scala/co/ledger/lama/bitcoin/worker/App.scala:24]: You should refactor the client creation out.

**Pull request merged**